contrib/babelfishpg_tsql/src/tsql_for/forxml.c

```diff
@@ -283,10 +332,33 @@ tsql_row_to_xml_path(StringInfo state, Datum record, const char *element_name, b
 		if (!isnull)
 		{
 			allnull = false;
-			appendStringInfo(state, "<%s>%s</%s>",
-							 colname,
-							 map_sql_value_to_xml_value(colval, datatype_oid, true),
-							 colname);
+			if(NameStr(att->attname)[0] == '@')
+			{
+				appendStringInfo(state, "%s=\"%s\" ",
+								 NameStr(att->attname)+1,
+								 map_sql_value_to_xml_value(colval, datatype_oid, true));
+			}
+			else
+			{
+				if (hasAttCentric && first)
+				{
+					appendStringInfoChar(state, '>');
+					first = false;
+				}
+				if(strncmp(NameStr(att->attname), "?column?", 8) == 0)
+				{
+					/* Dont include Default Colname that is assigned by PG */
+					appendStringInfo(state, "%s",
+									map_sql_value_to_xml_value(colval, datatype_oid, true));
+				}
```

Comment:
What if column itself has same name, I mean it is not default colname and created like `create table xyz([?column?] int)`? Is this is valid usecase?

Reply:
I checked the following in T-SQL
```
1> create table test_nirmit (SequenceNumber  INTEGER PRIMARY KEY, [?column?] VARCHAR(100) NOT NULL)
2> go
1> insert into test_nirmit values (1,'SELECT'),(2,'Product,'),(3,'UnitPrice,')
2> go

(3 rows affected)
1> SELECT [?column?]  FROM test_nirmit ORDER BY SequenceNumber FOR XML PATH
2> go
Msg 6850, Level 16, State 1, Line 1
Column name '?column?' contains an invalid XML identifier as required by FOR XML; '?'(0x003F) is the first character at fault.
1> 
```

This doesnt seem to be a supported column name to be used with XML PATH.

Reply:
And what happens in Babelfish?